### PR TITLE
Hardpoint UI Fixes

### PR DESCRIPTION
**Pull request by x3Karma**
### TL;DR:
- Fix the missing overhead UI mentioned in https://github.com/R2Northstar/NorthstarMods/issues/42.
- Added missing pulsating UI when a hardpoint is amped and decaying.
image

### Methodology: 
When using `SetHardpointState(CAPTURE_POINT_STATE_AMPING)`, the game will say that it is "CONTESTED" even though there's no enemy. The bar will still increase even though it said that its being contested.
image
This is replaced with an arbitrary number of 8 which removes the "CONTESTED" UI bug.
image

</br>

This shouldn't affect the actual contested UI from being displayed as the code checks for contested zones first before displaying amping UI.

### CODE REVIEW

**Comment by ASpoonPlaysGames** [comment]
Is there a chance that this is an underlying problem with sh_consts or something? Maybe we are missing another CAPTURE_POINT_STATE somehow?

**Comment by x3Karma** [comment]
> Is there a chance that this is an underlying problem with sh_consts or something? Maybe we are missing another CAPTURE_POINT_STATE somehow?

Could be, otherwise I have no clue how this works. Past the number 8, every subsequent even number would put it back to normal "Amping _%" but every odd number puts it back to "CONTESTED"

**Review by EladNLG — APPROVED** [review]

**Review by ASpoonPlaysGames — APPROVED** [review]

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut [hunk 1]
```diff
@@ -546,8 +548,10 @@ void function HardpointThink( HardpointStruct hardpoint )
 			}
 			else if(file.ampingEnabled)//amping or reamping
 			{
-				if(GetHardpointState(hardpoint)<CAPTURE_POINT_STATE_AMPING)
-				SetHardpointState(hardpoint,CAPTURE_POINT_STATE_AMPING)
+				// i have no idea why but putting it CAPTURE_POINT_STATE_AMPING will say 'CONTESTED' on the UI
+				// since whether the point is contested is checked above, putting the hardpoint state to a value of 8 fixes it somehow
+				if(GetHardpointState(hardpoint)<=CAPTURE_POINT_STATE_AMPING)
+					SetHardpointState(hardpoint, 8 )
```
Comment (EladNLG) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut:554]: Question: Why do we have to do this?

Like, it doesn't happen in vanilla, why does it in Northstar?

IMO, we should get to the core of the issue, not just put a band aid over it.

**Review by EladNLG — CHANGES_REQUESTED** [review]

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 1 above)
Comment (x3Karma) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut:554]: It being 3 months ago, I don't have a strong memory but I recall that `sh_consts` does not have the value for amping state I think.

**Review by NoCatt — APPROVED** [review]
Fixes work on test server, in a all contesting situations.

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 1 above)
Comment (GeckoEidechse) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut:554]: Wouldn't that imply some higher level issue in Northstar?

**Comment by GeckoEidechse** [comment]
Merging as soon as I get a recent code review ^^

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 1 above)
Comment (EladNLG) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut:554]: It would, I guess. I don't think this warrants a merge block though.

It works.

Comment (EladNLG) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut]: consistent spacing, please.

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut [hunk 2]
```diff
@@ -662,6 +668,8 @@ void function OnHardpointLeft( entity trigger, entity player )
 	foreach(CP_PlayerStruct playerStruct in file.players)
 		if(playerStruct.player == player)
 			playerStruct.isOnHardpoint = false
+		
+	player.SetPlayerNetInt( "playerHardpointID", 69) // an arbitary number to remove the hud from the player
```
Comment (EladNLG) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut:672]: ugh.

**Review by EladNLG — APPROVED** [review]
I like when the `playerHardpointID` is `69` :)

Anyways, took a look over everything, and it looks good - no real complaints.

We should, however, turn the `SetHardpointState( hardpoint, 8 )` thing into an issue because I will not accept hacks in my Titanfall 2 Northstar mod >:(((

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 1 above)
Comment (EladNLG) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut]: (resolve this is outdated GAH)

**Comment by EladNLG** [comment]
(gecko I forgot to remove the conversations i cannot delete them please dont hurt me)

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 2 above)
Comment (x3Karma) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut]: foixed

Comment (EladNLG) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut]: mark this as resolved then

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut (hunk 1 above)
Comment (GeckoEidechse) [reply Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut]: Conversation moved to https://github.com/R2Northstar/NorthstarMods/issues/601

**Review by GeckoEidechse — APPROVED** [review]
I did not test / code review PR, simply giving sign-off approval based on other reviews.

**Comment by pg9182** [comment]
```
[22:34:11] [NATIVE SV] [info] Player pg9182 fully connected at time 5.73333
[22:34:11] [SCRIPT SV] [info] Player client script initialization complete: entity (1: player pg9182 [1])
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] CommandString: ClientStatus was not added via AddClientCommandCallback but is being called in CodeCallback_ClientCommand
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] CommandString: AllDialogueFinished was not added via AddClientCommandCallback but is being called in CodeCallback_ClientCommand
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [NATIVE SV] [info] Setting player "player pg9182 [1]" predicted first person proxy model to "" (class "spectator")
[22:34:11] [SCRIPT SV] [info] done waiting!
[22:34:12] [SCRIPT SV] [info] started intro!
[22:34:12] [SCRIPT SV] [info] starting dropship intro!
[22:34:12] [SCRIPT SV] [info] SCRIPT ERROR: [SERVER] Entity class "CNPC_Dropship" doesn't match required class for this index or function call
[22:34:12] [SCRIPT SV] [info]  -> player.SetPlayerNetInt( "playerHardpointID", hardpoint.hardpoint.GetHardpointID() )
[22:34:12] [SCRIPT SV] [info]
CALLSTACK
*FUNCTION [OnHardpointEntered()] gamemodes/_gamemode_cp.nut line [654]

[22:34:12] [SCRIPT SV] [info] LOCALS
[hardpoint] unknown struct
[player] ENTITY (npc_dropship [128] (NPC "npc_dropship" at <933.302 -1869.51 314.585>))
[trigger] ENTITY (trigger_capture_point trigger_hardpoint_c [203] at <1404 -1940 476>)
[this] TABLE

DIAGPRINTS
```

**Review by pg9182 — COMMENTED** [review]
Crashed when applied on 1.12 and testing with just myself in private match / cp / mp_colony02:

```
[22:34:11] [NATIVE SV] [info] Player pg9182 fully connected at time 5.73333
[22:34:11] [SCRIPT SV] [info] Player client script initialization complete: entity (1: player pg9182 [1])
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] CommandString: ClientStatus was not added via AddClientCommandCallback but is being called in CodeCallback_ClientCommand
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [SCRIPT SV] [info] CommandString: AllDialogueFinished was not added via AddClientCommandCallback but is being called in CodeCallback_ClientCommand
[22:34:11] [SCRIPT SV] [info] ############################
[22:34:11] [NATIVE SV] [info] Setting player "player pg9182 [1]" predicted first person proxy model to "" (class "spectator")
[22:34:11] [SCRIPT SV] [info] done waiting!
[22:34:12] [SCRIPT SV] [info] started intro!
[22:34:12] [SCRIPT SV] [info] starting dropship intro!
[22:34:12] [SCRIPT SV] [info] SCRIPT ERROR: [SERVER] Entity class "CNPC_Dropship" doesn't match required class for this index or function call
[22:34:12] [SCRIPT SV] [info]  -> player.SetPlayerNetInt( "playerHardpointID", hardpoint.hardpoint.GetHardpointID() )
[22:34:12] [SCRIPT SV] [info]
CALLSTACK
*FUNCTION [OnHardpointEntered()] gamemodes/_gamemode_cp.nut line [654]

[22:34:12] [SCRIPT SV] [info] LOCALS
[hardpoint] unknown struct
[player] ENTITY (npc_dropship [128] (NPC "npc_dropship" at <933.302 -1869.51 314.585>))
[trigger] ENTITY (trigger_capture_point trigger_hardpoint_c [203] at <1404 -1940 476>)
[this] TABLE

DIAGPRINTS
```

Doesn't crash before this PR.
